File: python/exec/export_unknown_wiki_count_list_by_namespace.py
import sys
import os
import shutil
import glob
sys.path.append('./src')

from unknown_wiki_count_list_exporter import UnknownWikiCountListExporter

_, group_by, language, *_ = sys.argv

print('-------------------- Exporting Unknown Wiki Count List... --------------------')
match group_by:
    case '-o' | '--owner' | '-c' | '--category':
        count_list_by_namespace, path_to_export = UnknownWikiCountListExporter(group_by = group_by).run()

        match language:
            case '-en' | '-ja':
                count_list_by_namespace, path_to_export = UnknownWikiCountListExporter(group_by = group_by, language = language).run()
    case _:
        count_list_by_namespace, path_to_export = UnknownWikiCountListExporter().run()
print()
print('Here is the result:')
print()
print('---------------------------------------')
print(count_list_by_namespace.rstrip())
print('---------------------------------------')
print()
print(f"Check it out result on '{path_to_export}' !!")
print()
print('-------------------- Done Exporting Unknown Wiki Count List 🎉 --------------------')

for pycache in glob.glob(os.path.join('.', '**', '__pycache__'), recursive = True):
    if os.path.exists(pycache):
        shutil.rmtree(pycache)

File: ruby/exec/export_unknown_wiki_count_list_by_namespace.rb
require_relative '../src/unknown_wiki_count_list_exporter'

group_by, language, *_ = ARGV
params = { group_by:, language: }.reject { |_, value| value.empty? }

puts '-------------------- Exporting Unknown Wiki Count List... --------------------'
count_list_by_namespace, path_to_export = UnknownWikiCountListExporter.run(**params)
puts "\n"
puts 'Here is the result:'
puts "\n"
puts '---------------------------------------'
count_list_by_namespace.each { |count_list|
  puts count_list
}
puts '---------------------------------------'
puts "\n"
puts "Check it out result on '#{path_to_export}' !!"
puts "\n"
puts '-------------------- Done Exporting Unknown Wiki Count List 🎉 --------------------'
